core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -460,7 +463,15 @@ public ByteBuffer keyMetadata() {
 
   @Override
   public List<Long> splitOffsets() {
-    return ArrayUtil.toLongList(splitOffsets);
+    if (splitOffsetsAsList == null && splitOffsets != null) {
```

Comment:
This method is called at least 3 times and converting the array on each call is a bad idea.

<img width="2559" alt="image" src="https://github.com/apache/iceberg/assets/6235869/f708d3ac-771d-4f2b-828d-2ab590418163">

Reply:
Maybe, even a better idea would be to access the underlying offset array and use that directly. Let me give that a try, will get rid of a lot of boxing/unboxing in `OffsetsAwareSplitScanTaskIterator`.